devito/symbolics/manipulation.py

```diff
@@ -222,6 +222,8 @@ def as_symbol(expr):
         return Symbol(expr)
     elif isinstance(expr, Dimension):
         return Symbol(expr.name)
+    elif (isinstance(expr.args[0], Number) and isinstance(expr.args[1], Dimension)):
```

Comment:
@FabioLuporini what about having that now instead of Mul? Do you think it is better now?